bin/smarthome.py

```diff
@@ -940,8 +955,23 @@ def _reload_logics():
     arggroup.add_argument('-q', '--quiet', help='DEPRECATED use logging config (reduce logging to the logfile)', action='store_true')
     arggroup.add_argument('-V', '--version', help='show SmartHomeNG version', action='store_true')
     arggroup.add_argument('--start', help='start SmartHomeNG and detach from console (default)', default=True, action='store_true')
+    argparser.add_argument('-c', '--config_dir', help='use external config dir (should contain "etc", "logics" and "items" subdirectories)')
     args = argparser.parse_args()
 
+    extern_conf_dir = None
+    if args.config_dir is not None:
+        if os.path.isdir(args.config_dir):
+            missing = []
+            for d in ['etc', 'logics', 'items']:
+                if not os.path.isdir(os.path.join(args.config_dir, d)):
+                    missing.append(d)
+            if not missing:
+                extern_conf_dir = os.path.normpath(args.config_dir)
+            else:
+                print("ERROR: missing {} subdirectory(ies) in {}. Looking for config in {}".format(missing, args.config_dir, BASE))
+        else:
+            print("ERROR: {} is not a directory. Looking for config in {}".format(args.config_dir, BASE))
+
```

Comment:
I think we can just list the missing files and do not explicitly check the if the configuration directory exists. The user should be able to see that the directory is missing/wrong when SHNG is reporting missing configuration files.

Reply:
Just something like this?
```
extern_conf_dir = BASE
if args.config_dir is not None:
    extern_conf_dir = os.path.normpath(args.config_dir)
```

Reply:
Yes, like this.